src/backend/api/meals.js

```diff
@@ -2,14 +2,213 @@ const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
+// /api/meals	GET	Returns all meals
+
 router.get("/", async (request, response) => {
+  let query = knex("meal");
+
+  if (
+    "maxPrice" in request.query &&
+    typeof request.query.maxPrice === "string"
+  ) {
+    query = query.whereRaw("price <= ?", [request.query.maxPrice]);
+  }
+
+  if (request.query.availableReservations) {
+    query = query.join("reservation", "meal.id", "=", "reservation.meal_id");
+  }
+
+  if ("title" in request.query && typeof request.query.title === "string") {
+    query = query.whereRaw("title LIKE ?", [`%${request.query.title}%`]);
+  }
+
+  if (
+    "dateAfter" in request.query &&
+    typeof request.query.dateAfter === "string"
+  ) {
+    query = query.whereRaw("`when` > ?", [request.query.dateAfter]);
+  }
+
+  if (
+    "dateBefore" in request.query &&
+    typeof request.query.dateBefore === "string"
+  ) {
+    query = query.whereRaw("`when` > ?", [request.query.dateBefore]);
+  }
+
+  if ("limit" in request.query && typeof request.query.limit === "string") {
+    query = query.limit(`${request.query.limit}`);
+  }
+
+  if (request.query.sort_key && request.query.sort_dir) {
+    const sortAllowed = ["when", "max_reservations", "price"];
+    const orderBy = request.query.sort_key.toString();
+    let queryIncludesAllowedSort = sortAllowed.some((element) =>
+      orderBy.includes(element)
+    );
+    if (queryIncludesAllowedSort) {
+      query = query.orderByRaw(`${orderBy} ${request.query.sort_dir}`);
+    } else {
+      response
+        .status(404)
+        .send("Bad request, only when, max_reservations, price allowed");
+    }
+  }
+
+  if (request.query.sort_key && !request.query.sort_dir) {
```

Comment:
I don't think you need to duplicate this entire block. If `sort_dir` is provided, use that, if not you can just assume it is `ASC`.

Reply:
did this ` if (request.query.sort_key) {
    const sortAllowed = ["when", "max_reservations", "price"];
    const orderBy = request.query.sort_key.toString();
    let queryIncludesAllowedSort = sortAllowed.some((element) =>
      orderBy.includes(element)
    );
    if (queryIncludesAllowedSort && request.query.sort_dir) {
      query = query.orderBy(orderBy, request.query.sort_dir);
    } else if (queryIncludesAllowedSort && !request.query.sort_dir) {
      query = query.orderByRaw(orderBy);
    } else {
      return response
        .status(404)
        .send("Bad request, only when, max_reservations, price allowed");
    }
  }`